screencaptor/src/main/java/com/wealthfront/screencaptor/ScreenCaptor.kt

```diff
@@ -80,50 +81,82 @@ object ScreenCaptor {
     screenshotFormat: ScreenshotFormat = PNG,
     screenshotQuality: ScreenshotQuality = BEST
   ) {
+    rootViews.forEach { view ->
+      Log.d(SCREENSHOT, "Root view has height(${view.height}) and width(${view.width})")
+      if (view.width == 0 || view.height == 0) {
+        throw IllegalStateException("This view ($view) has no height or width. Is ${view.id} the currently displayed activity?")
+      }
+    }
 
-    val device = MANUFACTURER.replaceWithUnderscore() + "_" + MODEL.replaceWithUnderscore()
+    val deviceName = MANUFACTURER.replaceWithUnderscore() + "_" + MODEL.replaceWithUnderscore()
+    val screenshotId = "${screenshotName.toLowerCase(ENGLISH)}_${deviceName}_${SDK_INT}_$screenshotNameSuffix"
+    val screenshotFilePath = "$screenshotDirectory/$screenshotId.${screenshotFormat.extension}"
 
-    val screenshotName = "${screenshotFilename.toLowerCase(ENGLISH)}_${device}_${SDK_INT}_$screenshotNameSuffix"
-    val screenshotFile = "$screenshotDirectory/$screenshotName.${screenshotFormat.extension}"
+    val bitmap = createCanvasBitmap(rootViews)
+    val canvas = Canvas(bitmap)
 
-    Log.d(SCREENSHOT, "Posting to main thread for '$screenshotFile'")
     mainHandler.post {
-      Log.d(SCREENSHOT, "Root view has height(${view.height}) and width(${view.width})")
-      if (view.width == 0 || view.height == 0) {
-        throw IllegalStateException("This view ($view) has no height or width. Is ${view.id} the currently displayed activity?")
+      Log.d(SCREENSHOT, "Posting to main thread for '$screenshotId'")
+
+      rootViews.forEach { rootView ->
+        renderViewToCanvas(
+          canvas,
+          rootView,
+          viewIdsToExclude,
+          viewModifiers,
+          viewDataProcessor,
+          viewMutators,
+          screenshotName
+        )
       }
 
-      Log.d(SCREENSHOT, "Mutating views as needed!")
-      ViewTreeMutator.Builder()
-        .viewMutators(viewMutators)
-        .mutateView(view)
-        .mutate()
-
-      Log.d(SCREENSHOT, "Disabling views: $viewIdsToExclude")
-      val initialStateOfViews = ViewVisibilityModifier.hideViews(view, viewIdsToExclude)
-      val initialDataOfViews = viewDataProcessor.modifyViews(view, viewModifiers)
-
-      view.post {
```

Comment:
call out: in restructuring this code to iterate over multiple views, I experimented with dropping this and everything seemed to work identically. The overall code block is still wrapped in a `mainHandler.post`, which ensures that Espresso will wait for it to finish